<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>784</width>
    <height>605</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#lineEdit[state = &quot;0&quot;] {
	background-color: rgb(255, 0, 0);
}
#lineEdit[state = &quot;1&quot;] {
	background-color: rgb(0, 255, 0);
}
#lineEdit_3[state = &quot;0&quot;] {
	background-color: rgb(255, 0, 0);
}
#lineEdit_3[state = &quot;1&quot;] {
	background-color: rgb(0, 255, 0);
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="verticalSpacing">
       <number>6</number>
      </property>
      <item row="4" column="0">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_9"/>
        </item>
        <item row="1" column="2">
         <widget class="QGroupBox" name="groupBox">
          <property name="toolTip">
           <string>If either of AND or OR or XOR or NOT is selected, this will be active</string>
          </property>
          <property name="statusTip">
           <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied to the rest of the bytes without custom mask</string>
          </property>
          <property name="title">
           <string>Custom bitmask</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <layout class="QFormLayout" name="formLayout">
             <property name="verticalSpacing">
              <number>0</number>
             </property>
             <item row="0" column="1">
              <widget class="QLabel" name="label_byteNumber">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Displays current bitmask of a specified byte</string>
               </property>
               <property name="statusTip">
                <string>Displays current bitmask of a specified byte</string>
               </property>
               <property name="text">
                <string>Byte#0</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="lineEdit_3">
               <property name="toolTip">
                <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied to the rest of the bytes without custom mask</string>
               </property>
               <property name="statusTip">
                <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied to the rest of the bytes without custom mask</string>
               </property>
               <property name="inputMask">
                <string>BBBBBBBB</string>
               </property>
               <property name="text">
                <string>00000000</string>
               </property>
               <property name="state" stdset="0">
                <string>1</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_8">
               <property name="toolTip">
                <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied to the rest of the bytes without custom mask</string>
               </property>
               <property name="statusTip">
                <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied to the rest of the bytes without custom mask</string>
               </property>
               <property name="text">
                <string>Bitmask</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QPushButton" name="pushButton_bitMaskButton">
               <property name="toolTip">
                <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied</string>
               </property>
               <property name="statusTip">
                <string>Setting bitmask for each byte. If  file starts modifying without setting bitmask to all the bytes, the global mask will be applied</string>
               </property>
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="toolTip">
           <string>If either of AND or OR or XOR or NOT is selected, this will be active</string>
          </property>
          <property name="statusTip">
           <string>If either of AND or OR or XOR or NOT is selected, this will be active</string>
          </property>
          <property name="title">
           <string>Global bitmask</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_14">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <property name="spacing">
              <number>6</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_7">
               <property name="toolTip">
                <string>If mask </string>
               </property>
               <property name="text">
                <string>Global bitmask</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit">
               <property name="inputMask">
                <string>BBBBBBBB</string>
               </property>
               <property name="text">
                <string>00000000</string>
               </property>
               <property name="state" stdset="0">
                <string>1</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Byte amount</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Byte amount</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QGroupBox" name="groupBox_5">
          <property name="toolTip">
           <string>If either of shift operands is chosen, this will be active</string>
          </property>
          <property name="statusTip">
           <string>If either of shift operands is chosen, this will be active</string>
          </property>
          <property name="whatsThis">
           <string/>
          </property>
          <property name="title">
           <string>Bit shift settings</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_21">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_20">
             <property name="spacing">
              <number>6</number>
             </property>
             <property name="bottomMargin">
              <number>36</number>
             </property>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="statusTip">
                <string>The number of bits by which the byte sequence will be shifted</string>
               </property>
               <property name="text">
                <string>Shift count</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_2">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Directory selection</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="QLineEdit" name="lineEdit_2">
             <property name="text">
              <string>C:/</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_selectDirectory">
             <property name="text">
              <string>Select Directory</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_iterateLabel">
            <property name="text">
             <string>Iterate:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton_iterateOnce">
            <property name="text">
             <string>One time</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup_2</string>
            </attribute>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>112</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton_iterateTimer">
            <property name="text">
             <string>For</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup_2</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_3">
            <property name="wrapping">
             <bool>false</bool>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>999</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
            <property name="displayIntegerBase">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_iterateSeconds">
            <property name="text">
             <string>seconds</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Delete base files after modifying?</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_deleteFiles">
            <property name="text">
             <string/>
            </property>
            <property name="iconSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="2" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Collision behaviour</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton_collisionFilename">
            <property name="text">
             <string>Change filename</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton_collisionModify">
            <property name="text">
             <string>Overwrite</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="4" column="1" colspan="2">
       <widget class="QGroupBox" name="groupBox_6">
        <property name="title">
         <string>Workspace</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_22">
         <item>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="2" column="1">
            <widget class="QLabel" name="label_workspaceLabelStatus">
             <property name="text">
              <string>Status:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_workspaceActTimer">
             <property name="text">
              <string>iterates once</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_workspaceInfoTimer">
             <property name="text">
              <string>Timer:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QLabel" name="label_workspaceStatus">
             <property name="text">
              <string>awaits..</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QGroupBox" name="groupBox_workspaceGroupBox">
             <property name="title">
              <string>Found files</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_23">
              <item>
               <widget class="QListWidget" name="listWidget_workspaceListWidget"/>
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="1" colspan="2">
            <widget class="QPushButton" name="pushButton_workspaceStart">
             <property name="text">
              <string>Start</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>14</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Operand settings</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <family>Consolas</family>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>File Selection Settings</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QRadioButton" name="radioButton_operandAnd">
          <property name="text">
           <string>AND</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup_3</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_operandOr">
          <property name="text">
           <string>OR</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup_3</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_operandXor">
          <property name="text">
           <string>XOR</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup_3</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_operandNot">
          <property name="toolTip">
           <string>Inverts all the bits in all the bytes within the file</string>
          </property>
          <property name="statusTip">
           <string>Inverts all the bits in all the bytes within the file</string>
          </property>
          <property name="text">
           <string>NOT</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup_3</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_operandlShift">
          <property name="toolTip">
           <string>Shifts the specified amount if bits within the specified amount of bytes</string>
          </property>
          <property name="statusTip">
           <string>Shifts the specified amount if bits within the specified amount of bytes</string>
          </property>
          <property name="text">
           <string>Left shift</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup_3</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_operandRshift">
          <property name="toolTip">
           <string>Shifts the specified amount if bits within the specified amount of bytes</string>
          </property>
          <property name="statusTip">
           <string>Shifts the specified amount if bits within the specified amount of bytes</string>
          </property>
          <property name="text">
           <string>Right shift</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">buttonGroup_3</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="1" rowspan="4" colspan="2">
       <widget class="QGroupBox" name="groupBox_7">
        <property name="font">
         <font>
          <family>Consolas</family>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="title">
         <string>Mask Settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="2" column="0">
          <widget class="QLineEdit" name="lineEdit_masksAdd">
           <property name="font">
            <font>
             <family>Consolas</family>
             <pointsize>9</pointsize>
             <bold>false</bold>
            </font>
           </property>
           <property name="maxLength">
            <number>32</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_masksAdd">
           <property name="font">
            <font>
             <family>Consolas</family>
             <pointsize>9</pointsize>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>Set Mask</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_maskSettingMaskLabel">
           <property name="text">
            <string>Current mask:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_maskSettingCurrentMask">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QTextEdit" name="textEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:8pt;&quot;&gt;*.txt - find every .txt &lt;br /&gt;hello* - find every file that starts with &amp;quot;hello&amp;quot;, e.g &amp;quot;helloworld.txt&amp;quot;&lt;br /&gt;*hello*.txt - find every file that has word &amp;quot;hello&amp;quot; in it and .txt file extention, e.g &amp;quot;worldhello.txt&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>784</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
  <buttongroup name="buttonGroup_2"/>
  <buttongroup name="buttonGroup_3"/>
 </buttongroups>
</ui>
